### PR TITLE
SQL-777: Implementing tag-triggered release upload to public S3

### PR DESCRIPTION
**Pull request by bucaojit**
Added instructions to `RELEASE.md`.

Tagged commits will be uploaded to the `translators-connectors-releases` bucket, non-tagged will be uploaded to `mciuploads`.

### CODE REVIEW

File: RELEASE.md
Comment (nbagnard) [inline RELEASE.md:37]: Maybe I missed something, but I don't see this in the evergreen script either.
The idea is to have plugin-version={placeholder} in the repo and have the evergreen script automatically update it to the correct tag version before building.

**Review by nbagnard — CHANGES_REQUESTED** [review]
Just one thing regarding the plugin version in the manifest. We can discuss if what I wrote was not clear or if it does not make sense for any reason.
Otherwise I think everything looks good 👍

File: RELEASE.md
Comment (bucaojit) [reply RELEASE.md:37]: Oh, I see.  I had the idea that we would manually update the version in the manifest.xml.  
How about for {placeholder} I put '0.0.0' so that the taco build will pass the xml validation.  And then if tag-triggered it will find/replace the value with the tag?  The non-tag-triggered will keep the placeholder '0.0.0'.

Comment (nbagnard) [reply RELEASE.md:37]: Yes, sounds good to me. 0.0.0 seems very reasonable as a place holder 👍 
How uptight is Tableau about the version format, do you know?
For non tag triggered, to differentiate them, maybe we could see how to add the git version after the patch number? I'll create an investigation ticket for this since it does not apply to this JIRA itself, but it's good to keep track of it.

File: evergreen.yml
Comment (nbagnard) [inline evergreen.yml:181]: Maybe we should check if the name works for Alexi. I checked the design doc and I did not specify the name of the taco file, so she might have missed that too.

File: RELEASE.md
Comment (nbagnard) [inline RELEASE.md:63]: Can you also add the URL to the README as well so that users who come from the repo have access to the released taco? I tried to describe this in the "Distribution" section of the spec, but I realize that it is a little succinct. I think for now, just a line saying something around the line "The releases are available for download here : https://translators-connectors-releases.s3.us-east-1.amazonaws.com/mongo-tableau-connector/mongodb-jdbc-<plugin-version>.taco"

**Review by nbagnard — CHANGES_REQUESTED** [review]

File: RELEASE.md
Comment (bucaojit) [reply RELEASE.md:37]: I experimented earlier with setting it to something like '0.0.0-78b19e4'.  It doesn't work as the validation check seems to want only digits in the version.  It checks against this pattern for validation: 
`['(\\d+)((\\.)(\\d+))*((-)(\\d+))*']`

File: evergreen.yml
Comment (bucaojit) [reply evergreen.yml:181]: Sounds good. Checking with Alexi.

File: RELEASE.md
Comment (bucaojit) [reply RELEASE.md:63]: Good catch, I added the URL to the README.

File: README.md
Comment (nbagnard) [inline README.md:8]: What about providing the bucket link only and omit the file name containing the plugin-version so that they can then choose which one they want?

File: RELEASE.md
Comment (nbagnard) [reply RELEASE.md:37]: ok, cool, it's nice you already checked. I'll add this as a comment on the related JIRA then. Maybe we can simply add the version as a suffix for the taco file name.

File: evergreen.yml
Comment (nbagnard) [reply evergreen.yml:181]: Great, thank you 😊

**Review by nbagnard — APPROVED** [review]
LGTM! I just have a small comment/question regarding the URL provided in the README that I leave to your best judgment.

File: README.md
Comment (bucaojit) [reply README.md:8]: Discussed offline and decided to leave as-is since the bucket link only would lead to an Access Denied page.

**Review by rychipman — APPROVED** [review]
code changes LGTM! once we merge this PR (but before closing the ticket), let's try to tag a release to make sure all of this works as expected with the tag-triggered versions

**Comment by bucaojit** [comment]
evergreen merge

**Pull request merged**
